pkg/controller/iampolicy/iampolicy_controller.go

```diff
@@ -346,9 +350,11 @@ func checkComplianceBasedOnDetails(plc *policiesv1.IamPolicy) {
 	}
 	for namespace, msgList := range plc.Status.CompliancyDetails[plc.Name] {
 		if len(msgList) > 0 {
-			violationNum := strings.Split(plc.Status.CompliancyDetails[plc.Name][namespace][0], " ")
-			if len(violationNum) >= 7 {
-				if violationNum[7] != fmt.Sprint(0) && strings.HasPrefix(violationNum[0], "Number") {
+			regexStr := fmt.Sprintf(violationMsgFUserCountRegex, regexp.QuoteMeta(roleName))
```

Comment:
Optional: It'd be nice to put this code in a helper function to avoid the duplication.